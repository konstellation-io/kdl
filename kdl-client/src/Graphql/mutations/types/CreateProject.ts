/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import {
  CreateProjectInput,
  RepositoryType,
  ProjectState,
} from './../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: CreateProject
// ====================================================

export interface CreateProject_createProject_repository {
  __typename: 'Repository';
  id: string;
  type: RepositoryType;
  url: string;
  connected: boolean;
}

export interface CreateProject_createProject {
  __typename: 'Project';
  id: string;
  name: string;
  description: string;
  favorite: boolean;
  creationDate: string;
  lastActivationDate: string;
  repository: CreateProject_createProject_repository | null;
  state: ProjectState;
  error: string | null;
}

export interface CreateProject {
  createProject: CreateProject_createProject;
}

export interface CreateProjectVariables {
  input: CreateProjectInput;
}
