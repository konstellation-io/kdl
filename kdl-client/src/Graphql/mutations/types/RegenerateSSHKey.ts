/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: RegenerateSSHKey
// ====================================================

export interface RegenerateSSHKey_regenerateSSHKey {
  __typename: 'SSHKey';
  public: string;
  private: string;
  creationDate: string;
  lastActivity: string | null;
}

export interface RegenerateSSHKey {
  regenerateSSHKey: RegenerateSSHKey_regenerateSSHKey;
}
