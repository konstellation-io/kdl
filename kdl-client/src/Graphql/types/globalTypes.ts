/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum AccessLevel {
  ADMIN = 'ADMIN',
  MANAGER = 'MANAGER',
  VIEWER = 'VIEWER',
}

export enum ProjectState {
  ARCHIVED = 'ARCHIVED',
  STARTED = 'STARTED',
  STOPPED = 'STOPPED',
}

export enum RepositoryType {
  EXTERNAL = 'EXTERNAL',
  INTERNAL = 'INTERNAL',
}

export interface AddUserInput {
  email: string;
  accessLevel: AccessLevel;
}

export interface CreateProjectInput {
  name: string;
  description: string;
  repository: RepositoryInput;
}

export interface RemoveUsersInput {
  userIds: string[];
}

export interface RepositoryInput {
  type: RepositoryType;
  url: string;
}

export interface UpdateAccessLevelInput {
  userIds: string[];
  accessLevel: AccessLevel;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
