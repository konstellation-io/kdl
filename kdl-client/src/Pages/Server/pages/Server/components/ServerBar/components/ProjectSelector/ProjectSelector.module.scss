@import 'Styles/variables';
@import 'Styles/mixins';
@import 'Styles/colors';

.container {
  z-index: 3;
  ul {
    list-style-type: none;
    padding: 0;
    margin: 0;
    .project {
      @include font-body;
      color: font-color(lowlight);
      padding: $grid-unit;
      &.selectedProject {
        background-color: palette(highlight);
        color: font-color(grey);
      }
    }
  }
}
