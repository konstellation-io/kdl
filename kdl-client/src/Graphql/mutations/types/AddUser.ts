/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { AddUserInput, AccessLevel } from './../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: AddUser
// ====================================================

export interface AddUser_addUser {
  __typename: 'User';
  id: string;
  email: string;
  accessLevel: AccessLevel;
  creationDate: string;
  lastActivity: string | null;
}

export interface AddUser {
  addUser: AddUser_addUser;
}

export interface AddUserVariables {
  input: AddUserInput;
}
