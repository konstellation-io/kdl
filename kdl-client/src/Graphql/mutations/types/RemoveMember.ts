/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { RemoveMemberInput } from './../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: RemoveMember
// ====================================================

export interface RemoveMember_removeMember {
  __typename: 'Member';
  id: string;
}

export interface RemoveMember {
  removeMember: RemoveMember_removeMember;
}

export interface RemoveMemberVariables {
  input: RemoveMemberInput;
}
