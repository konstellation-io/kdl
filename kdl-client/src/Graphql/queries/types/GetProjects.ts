/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { RepositoryType, ProjectState } from './../../types/globalTypes';

// ====================================================
// GraphQL query operation: GetProjects
// ====================================================

export interface GetProjects_projects_repository {
  __typename: 'Repository';
  id: string;
  type: RepositoryType;
  url: string;
  connected: boolean;
}

export interface GetProjects_projects {
  __typename: 'Project';
  id: string;
  name: string;
  description: string;
  favorite: boolean;
  creationDate: string;
  lastActivationDate: string;
  repository: GetProjects_projects_repository | null;
  state: ProjectState;
  error: string | null;
}

export interface GetProjects {
  projects: GetProjects_projects[];
}
