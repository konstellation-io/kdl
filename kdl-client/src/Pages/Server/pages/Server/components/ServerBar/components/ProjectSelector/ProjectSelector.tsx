import React, { FC } from 'react';
import styles from './ProjectSelector.module.scss';
import { GetProjects_projects } from 'Graphql/queries/types/GetProjects';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import ROUTE, { buildRoute } from 'Constants/routes';
import { Button } from 'kwc';
import IconAdd from '@material-ui/icons/Add';

type Props = {
  options: GetProjects_projects[];
  selectedProjectId: string;
  serverId: string;
};
const ProjectSelector: FC<Props> = ({
  selectedProjectId,
  options,
  serverId,
}) => (
  <div className={styles.container}>
    <ul>
      {options.map(({ id, name }: GetProjects_projects) => (
        <Link to={buildRoute.project(ROUTE.PROJECT, serverId, id)} key={id}>
          <li
            className={cx(styles.project, {
              [styles.selectedProject]: id === selectedProjectId,
            })}
          >
            {name}
          </li>
        </Link>
      ))}
    </ul>
    <Button
      Icon={IconAdd}
      label="NEW PROJECT"
      className={styles.addProjectButton}
      to={buildRoute.server(ROUTE.NEW_PROJECT, serverId)}
    />
  </div>
);
export default ProjectSelector;
