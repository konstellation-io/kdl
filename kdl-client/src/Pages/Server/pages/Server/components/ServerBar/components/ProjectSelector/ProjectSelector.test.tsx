import ProjectSelector from './ProjectSelector';
import React from 'react';
import { shallow } from 'enzyme';
import { GetProjects_projects } from 'Graphql/queries/types/GetProjects';
import { ProjectState } from '../../../../../../../../Graphql/types/globalTypes';

const project: GetProjects_projects = {
  id: 'foo',
  name: 'bar',
  __typename: 'baz',
  creationDate: 'foofoo',
  description: 'foobar',
  error: 'foobaz',
  favorite: true,
  lastActivationDate: 'barfoo',
  repository: null,
  state: ProjectState.STARTED,
};
const projects: GetProjects_projects[] = [project];

const props = {
  options: projects,
  selectedProjectId: project.id,
  serverId: 'bar',
};

let component;
beforeEach(() => {
  component = shallow(<ProjectSelector {...props} />);
});

describe('ProjectSelector component', () => {
  it('should render without crashing', function () {
    expect(component).toMatchSnapshot();
  });
});
