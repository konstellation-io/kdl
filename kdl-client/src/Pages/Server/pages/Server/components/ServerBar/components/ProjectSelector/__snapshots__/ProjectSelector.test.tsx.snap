// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSelector component should render without crashing 1`] = `
<div
  className="container"
>
  <ul>
    <Link
      key="foo"
      to="/server/bar/project/foo"
    >
      <li
        className="project selectedProject"
      >
        bar
      </li>
    </Link>
  </ul>
</div>
`;
